test/fixtures/typed_test.correct.rb

```diff
@@ -1032,4 +1058,47 @@ def to_h
     result["bytes_field".to_sym] = @bytes_field
     result
   end
+
+  private
+
+  sig { params(val: Integer).returns(Integer) }
+  def uint64_encoded_size(val)
+    if val > 0x7f
+      if val > 0x3fff
+        if val > 0x1fffff
+          if val > 0xfffffff
+            if val > 0x7ffffffff
+              if val > 0x3ffffffffff
+                if val > 0x1ffffffffffff
+                  if val > 0xffffffffffffff
+                    if val > 0x7fffffffffffffff
+                      10
+                    else
+                      9
+                    end
+                  else
+                    8
+                  end
+                else
+                  7
+                end
+              else
+                6
+              end
+            else
+              5
+            end
+          else
+            4
+          end
+        else
+          3
+        end
+      else
+        2
+      end
+    else
+      1
+    end
```

Comment:
```suggestion
    val.bit_length / 7
```
Why not? I did a quick bench and it seems to be very much the same performance. Also it would probably be easy for YJIT to optimize `Integer#bit_length` into something using `__builtin_clz ` or similar.

Reply:
sure, makes sense.  We need to fix the code generator though